### DISCUSSION TITLE
Why seed_everything() can't eliminate uncertainty?

### DISCUSSION THREAD

**Discussion by wuyucheng2002** [pyg-team/pytorch_geometric#3821] [Q&A] [answered]
Hello, I use seed_everything() function of pyg in the front of my GCN code, and if I run 100 epochs the results are the same, but if I run 500 epochs the results are always different and have little deviation (like the deviation between0.9269 and 0.9260). I am confused how the little error come about, is this because of the accumulation of errors in computer calculations? Has that ever happened to you? And how to solve the uncertainty of code recurrence? I'd appreciate it if you could answer my question.

**Comment by rusty1s** [comment, marked as answer]
Yes, there is an underlying non-determinism in the `scatter_*` functions that cannot be eliminated. See https://github.com/pyg-team/pytorch_geometric/issues/3175 for a more detailed discussion.
